src/app/core/services/user-profile.service.js

```diff
@@ -22,17 +22,26 @@ export default function (app) {
         $rootScope.currentUserId = uid;
       };
 
-      this.createFormInfo = function (rootScope) {
+      this.createFormInfo = function () {
         return {
-          firstName: rootScope.currentUser.firstName,
-          lastName:  rootScope.currentUser.lastName,
-          role: rootScope.currentUser.role,
-          phone:  rootScope.currentUser.phone,
-          email:  rootScope.currentUser.email,
-          ava: rootScope.currentUser.ava
+          firstName: $rootScope.currentUser.firstName,
+          lastName:  $rootScope.currentUser.lastName,
+          role: $rootScope.currentUser.role,
+          phone:  $rootScope.currentUser.phone,
+          email:  $rootScope.currentUser.email,
+          ava: $rootScope.currentUser.ava
         }
       };
 
+      this.saveToCurrentUser = function (data) {
+        $rootScope.currentUser.firstName = data.firstName ? data.firstName : '',
```

Comment:
Let's replace this form:
$rootScope.currentUser.firstName = data.firstName ? data.firstName : ''
with this form
$rootScope.currentUser.firstName = data.firstName || ''